src/app/sso/page.tsx

```diff
@@ -0,0 +1,44 @@
+import { cookies } from "next/headers";
+import { redirect } from "next/navigation";
+
+type Props = {
+  searchParams: { sso: string; sig: string };
+};
+
+const BASE_URL = "https://training.olinfo.it";
+
+export default async function Page({ searchParams: { sso: payload, sig: signature } }: Props) {
+  const token = cookies().get("training_token")?.value;
+
+  // User is logged out.
+  if (token === undefined) {
+    const redirectUrl = `/sso?sso=${payload}&sig=${signature}`;
+    redirect(`${BASE_URL}/login?redirect=${encodeURIComponent(redirectUrl)}`);
+  }
+
+  // TODO: need to implement this in @olinfo/training-api, to avoid hardcoding
+  // the BASE_URL here.
+  const res = await fetch(`${BASE_URL}/api/sso`, {
+    method: "POST",
+    headers: {
+      "Content-Type": "application/json",
+      Cookie: `training_token=${token}`,
+    },
+    body: JSON.stringify({
+      payload,
+      sig: signature,
+    }),
+  });
+
+  if (!res.ok) {
+    throw Error("Error querying Discourse SSO");
+  }
+
+  const { parameters } = await res.json();
+
+  // TODO: we need to get the forum_url from the contest, instead of hardcoding.
```

Comment:
Once https://github.com/algorithm-ninja/cmsocial/pull/37 is merged this should be easier